pkg/cli/cmds/agent.go

```diff
@@ -220,6 +221,11 @@ var (
 		Usage:       "(agent/networking) (experimental) Disable the agent's client-side load-balancer and connect directly to the configured server address",
 		Destination: &AgentConfig.DisableLoadBalancer,
 	}
+	DisableDefaultRegistryEndpointFlag = &cli.BoolFlag{
```

Comment:
Why would a user want to activate this? Isn't it good to have a fallback?

Reply:
As a user... I really like this flag if I understand what it does correctly.

This would make my life easier when air-gapping workloads, making testing that I have mirrored all of the required containers easier.

From a security perspective, I don't like my production non air-gapped clusters pulling untrusted containers down from external registries. They should only be running known trusted images, from trusted local sources.

I would use this flag on all production systems, and for some testing/development work.

Reply:
We've had folks open issues about it in the past, expressing surprise that containerd was falling back to the default public endpoint despite having a local mirror configured. Now that we have an option to disable this on the containerd side, I figured it'd be nice to expose that here in k3s.